src/Controls/src/Core/Application/Application.Windows.cs

```diff
@@ -1,6 +1,132 @@
-﻿namespace Microsoft.Maui.Controls
+﻿using System;
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.UI.Windowing;
+using Microsoft.UI.Xaml;
+
+namespace Microsoft.Maui.Controls
 {
 	public partial class Application
 	{
+		AppTheme? _currentThemeForWindows;
+
+		partial void OnRequestedThemeChangedPlatform(AppTheme newTheme)
+		{
+			_currentThemeForWindows = newTheme;
+
+			if (!AnyWindowReady())
+			{
+				return;
+			}
+
+			ApplyThemeToAllWindows(newTheme, UserAppTheme == AppTheme.Unspecified);
+		}
+
+		partial void OnPlatformWindowAdded(Window window)
+		{
+			window.HandlerChanged += OnWindowHandlerChanged;
+
+			if (_currentThemeForWindows is not null && window.Handler is not null)
+			{
+				TryApplyTheme();
+			}
+		}
+
+		partial void OnPlatformWindowRemoved(Window window)
+		{
+			window.HandlerChanged -= OnWindowHandlerChanged;
+		}
+
+		void OnWindowHandlerChanged(object? sender, EventArgs e) => TryApplyTheme();
+
+		void TryApplyTheme()
+		{
+			if (!AnyWindowReady())
+			{
+				return;
+			}
+
+			if (_currentThemeForWindows is AppTheme theme)
+			{
+				ApplyThemeToAllWindows(theme, UserAppTheme == AppTheme.Unspecified);
+			}
+		}
+
+		bool AnyWindowReady()
+		{
+			foreach (var window in Windows)
+			{
+				var platformWindow = window?.Handler?.PlatformView as UI.Xaml.Window;
+				if (platformWindow?.Content is FrameworkElement)
+				{
+					return true;
+				}
+			}
+			return false;
+		}
+
+		void ApplyThemeToAllWindows(AppTheme newTheme, bool followSystem)
+		{
+			var forcedElementTheme = newTheme switch
```

Comment:
Can optimize it saving the latest applied theme and skip this if theme hasn't actually changed.

Reply:
The method is invoked either when the app theme changes or when the window handler changes (i.e., a new window is created).

For a new theme, it will always differ from the previous theme, so updates will be applied as needed.

For new windows, the method exits early if the current theme already matches the latest theme, since no updates are required. 

Please let me know if I have missed anything or if you have additional suggestions